PIL/WebPImagePlugin.py

```diff
@@ -30,17 +79,115 @@ class WebPImageFile(ImageFile.ImageFile):
+        lossy = True
+
+        first_chunk = True
+
+        while True:
+
+            chunk_header = self.fp.read(8)
+            if 8 != len(chunk_header):
+                if first_chunk:
+                    raise SyntaxError("not a WebP file")
+                break
+
+            chunk_fourcc = chunk_header[0:4]
+            chunk_size = i32le(chunk_header[4:8])
+
+            if first_chunk:
+                if chunk_fourcc not in _VP8_MODES_BY_IDENTIFIER:
+                    raise SyntaxError("not a WebP file")
+
+            if b'VP8X' == chunk_fourcc:
+                if first_chunk:
+                    mode, size, chunk_size = _parse_vp8x(chunk_size, self.fp)
+
+            elif b'VP8L' == chunk_fourcc:
+                if first_chunk:
+                    mode, size, chunk_size = _parse_vp8l(chunk_size, self.fp)
+                lossy = False
+
+            elif b'VP8 ' == chunk_fourcc:
+                if first_chunk:
+                    mode, size, chunk_size = _parse_vp8(chunk_size, self.fp)
+                lossy = True
+
+            elif b'ALPH' == chunk_fourcc:
+                mode = 'RGBA'
+
+            elif b'EXIF' == chunk_fourcc:
+                exif = self.fp.read(chunk_size)
+                if chunk_size != len(exif):
+                    raise SyntaxError("bad EXIF chunk")
+                self.info["exif"] = exif
+                chunk_size = 0
+
+            elif b'ICCP' == chunk_fourcc:
+                icc_profile = self.fp.read(chunk_size)
+                if chunk_size != len(icc_profile):
+                    raise SyntaxError("bad ICCP chunk")
+                self.info["icc_profile"] = icc_profile
+                chunk_size = 0
+
+            if chunk_size > 0:
+                # Skip to next chunk.
+                pos = self.fp.tell()
+                self.fp.seek(chunk_size, os.SEEK_CUR)
+                if self.fp.tell() != (pos + chunk_size):
+                    raise SyntaxError("not a WebP file")
+
+            first_chunk = False
+
+        if None in (mode, size, lossy):
+            raise SyntaxError("not a WebP file")
+
+        self.mode = mode
+        self.size = size
+        self.info['compression'] = 'lossy' if lossy else 'lossless'
+        self.tile = [('webp', (0, 0) + size, 0,
+                      # Decoder params: rawmode, has_alpha, width, height.
+                      (mode, 1 if 'RGBA' == mode else 0, size[0], size[1]))]
+
+        if not lossy:
+            # Incremental decoding on lossless is *really* slow, disable it.
+            self.decodermaxblock = 12 + i32le(header[4:8])
+            self.decoderconfig = (1,)
+
+    def load_prepare(self):
+        if not self.im:
+            # Hackety hack hack hack...
+            im = Image.core.new(self.mode, (0,0))
+            self.im = im.new_block(self.mode, self.size)
+        ImageFile.ImageFile.load_prepare(self)
+
+    def draft(self, mode, size):
+
+        if 1 != len(self.tile):
+            return
```

Comment:
I looked at `JpegImageFile.draft` for reference, and the code is the same: no `return self` in this case.